### PR TITLE
Define key environment variables

### LINKED ISSUE #5: Key enviroment variables not defined
When devboxes spin up (either after a scale down and back up, or new ones are spun up for new engineers), key environment variables are no longer defined. These include `HADOOP_CONF_DIR` and other hadoop-ecosystem variables.

Related: Factual/helpdesk#3158

### PR DESCRIPTION
**Pull request by cwienberg**
For reasons I couldn't figure out, devboxes don't have any of the hadoop environment variables defined for users anymore. I didn't see anything different on the dev and devbox images, or in the start scripts in resources, that would cause this.

This change is a bit hacky, in that it grabs *all* the environment variables defined in the Dockerfile inheritance chain and then blacklists a few. I don't like the approach, so am willing to hear suggestions; it just seemed like the best way to handle people defining key environment variables in (say) the docker-cdh5-base image and them not making it to devbox users.

I also do some clean-up to reduce docker image sizes.

Closes #5 
Closes Factual/helpdesk#3158

### CODE REVIEW

**Comment by cwienberg** [comment]
Here are the environment variables that will be placed in `/etc/environment`:

```
HADOOP_CONF_DIR=/etc/hadoop/conf
CDH_VERSION=5.13
MAVEN_PATH=/opt/apache-maven
SPARK_VERSION=2.3.1
PATH=/opt/hive/bin:/opt/spark/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PRESTO_VERSION=0.190
DEBIAN_FRONTEND=noninteractive
MAVEN_VERSION=3.5.2
HIVE_VERSION=2.3.2
THRIFT_VERSION=0.9.2
```

**Comment by nfultz** [comment]
`DEBIAN_FRONTEND` shouldn't be set for regular users - it disables the y/n prompt eg when the apt solver decides to replace glib or libc and hose the box .

**Comment by ndigati** [comment]
These environment variables are already available in the container at `/etc/container_environment.sh` because of [phusion/baseimage](https://github.com/phusion/baseimage-docker#environment-variables).

It might be a better idea to just `chgrp` the `/etc/container_environment.sh` file so those variables get added to the user environment automatically, since that script is sourced as part of the login process already (as long as the user has read access to that file)

We can probably just add a new line to the devbox start script similar to [this one](https://github.com/Factual/resources/blob/master/services/hadoop/cdh5/scripts/devbox.sh#L3):

```bash
chgrp "$MARATHON_APP_LABEL_OWNER" /etc/container_environment.sh
```

But we probably don't want all of these environment variables in users sessions. You probably only need ones like `PATH`, `MAVEN_PATH`, and `HADOOP_CONF_DIR`, so we might want to just write the ones we care about into something like `/etc/environment` or `/etc/profile.d/devbox_env.sh` in the Dockerfile

**Comment by cwienberg** [comment]
@ndigati You're right, we don't really need all of them. I think what we want are: `HADOOP_CONF_DIR, HIVE_HOME,MAVEN_PATH, PATH,SPARK_HOME` (hive and spark home weren't listed above). I'm cool with only writing those ones into `/etc/environment` or wherever else.

I still don't know why previous versions of the devbox image had these environment variables available when you SSHed in but now it does not.

**Comment by cwienberg** [comment]
@ndigati Do you have a better suggestion for this than my most recent change? If not, I think we should merge. It's already affecting [other engineers](https://factual.slack.com/archives/C0CU8H0HH/p1534357736000100).

File: Dockerfile [hunk 1]
```diff
@@ -4,19 +4,28 @@ RUN rm -f /etc/service/sshd/down
...
 RUN echo "deb https://apt.dockerproject.org/repo ubuntu-xenial main" >> /etc/apt/sources.list.d/docker.list
 
-RUN apt-get update && apt-get install -y byobu docker-engine bash-completion oozie-client wget
+RUN apt-get update && apt-get install -y \
+    bash-completion \
+    byobu \
+    docker-engine \
+    oozie-client \
+    wget && \
+    apt-get clean && \
```
Comment (ndigati) [inline Dockerfile]: https://github.com/Factual/docker-cdh5-base/pull/6#discussion_r210402998

**Pull request merged**